enterprise/admin/views.py

```diff
@@ -291,132 +292,396 @@ def _handle_bulk_upload(cls, enterprise_customer, manage_learners_form, request,
+            pending.append(pending_user)
+
+        return successes, pending, failures
+
+    @classmethod
+    def send_messages(cls, http_request, message_requests):
+        """
+        Deduplicate any outgoing message requests, and send the remainder.
+
+        Args:
+            http_request: The HTTP request in whose response we want to embed the messages
+            message_requests: A list of undeduplicated messages in the form of tuples of message type
+                and text- for example, ('error', 'Something went wrong')
+        """
+        deduplicated_messages = set(message_requests)
+        for msg_type, text in deduplicated_messages:
+            getattr(messages, msg_type)(http_request, text)
+
+    @classmethod
+    def course_notify(cls, enterprise_customer, request, course_id, users):
+        """
+        Notify learners about a course in which they've been enrolled.
+
+        Args:
+            enterprise_customer: The EnterpriseCustomer being linked to
+            request: The HTTP request that's being processed
+            course_id: The specific course the learners were enrolled in
+            users: An iterable of the users or pending users who were enrolled
+        """
+        course_details = CourseCatalogApiClient(request.user).get_course_run(course_id)
         if not course_details:
             logging.warning(
                 _(
                     "Course details were not found for course key {} - Course Catalog API returned nothing. "
                     "Proceeding with enrollment, but notifications won't be sent"
                 ).format(course_id)
             )
-            notify = False
-
-        if notify:
-            # Prefetch course metadata for drafting an email if we're going to send a notification
-            course_url = course_details.get('marketing_url')
-            if course_url is None:
-                # If we didn't get a useful path to the course on a marketing site from the catalog API,
-                # then we should build a path to the course on the LMS directly.
-                course_url = get_reversed_url_by_site(
-                    request,
-                    enterprise_customer.site,
-                    'about_course',
-                    args=(course_id,),
+            return
+        course_url = course_details.get('marketing_url')
+        if course_url is None:
+            # If we didn't get a useful path to the course on a marketing site from the catalog API,
+            # then we should build a path to the course on the LMS directly.
+            course_url = get_reversed_url_by_site(
+                request,
+                enterprise_customer.site,
+                'about_course',
+                args=(course_id,),
+            )
+        course_name = course_details.get('title')
+        course_start = parse_lms_api_datetime(course_details.get('start'))
+
+        with mail.get_connection() as email_conn:
```

Comment:
This could be factored out further which would open the door for triggering different types of notification mechanisms, such as SMS -- although it could be overkill for now.